### PR TITLE
Update the Golang bindings

### PR DESCRIPTION
**Pull request by dave-tucker**
This PR brings in changes from Docker for Mac and Windows to the Golang Datakit bindings.
This is necessary for us to switch to using the bindings upstream instead of our internal package

### CODE REVIEW

File: api/go-datakit/config.go [hunk 1]
```diff
@@ -178,14 +226,48 @@ func (f *Record) StringField(key string, value string) *StringField {
...
-	raw        *StringField
+	raw        *StringRefField
 	defaultInt int
 }
 
 // Get retrieves the current value of the key
 func (f *IntField) Get() (int, Version) {
-	value64, err := strconv.ParseInt(strings.TrimSpace(f.raw.value), 10, 0)
+	if f.raw.value == nil {
+		log.Printf("Key %s missing in database, defaulting value to to %t", strings.Join(f.raw.path, "/"), f.defaultInt)
```
Comment (talex5) [inline api/go-datakit/config.go]: Typo: `to to`.

File: api/go-datakit/config.go [hunk 2]
```diff
@@ -202,18 +284,22 @@ func (f *IntField) HasChanged(version Version) bool {
...
-	raw         *StringField
+	raw         *StringRefField
 	defaultBool bool
 }
 
 // Get retrieves the current value of the key
 func (f *BoolField) Get() (bool, Version) {
-	value, err := strconv.ParseBool(strings.TrimSpace(f.raw.value))
+	if f.raw.value == nil {
+		log.Printf("Key %s missing in database, defaulting value to to %t", strings.Join(f.raw.path, "/"), f.defaultBool)
```
Comment (talex5) [inline api/go-datakit/config.go]: `to to`

File: api/go-datakit/transaction_test.go [hunk 3]
```diff
@@ -24,7 +24,7 @@ func TestTransaction(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Transaction.Write failed: %v", err)
 	}
-	err = trans.Commit(ctx)
```
Comment (talex5) [inline api/go-datakit/transaction_test.go]: How did this work before?

Comment (dave-tucker) [inline api/go-datakit/transaction_test.go]: Signature was changed in 50d61ee but the tests weren't updated

Comment (samoht) [inline api/go-datakit/transaction_test.go]: we should probably run these tests in the CI ...

File: api/go-datakit/transaction.go [hunk 4]
```diff
@@ -96,12 +96,15 @@ func (t *transaction) Write(ctx context.Context, path []string, value string) er
 	return nil
 }
 
-// Remove a path (which can either be a key or a directory).
+// Remove deletes a key within the transaction.
 func (t *transaction) Remove(ctx context.Context, path []string) error {
 	p := []string{"branch", t.fromBranch, "transactions", t.newBranch, "rw"}
-	err := t.client.Remove(ctx, path...)
+	for _, dir := range path[0 : len(path)-1] {
```
Comment (samoht) [inline api/go-datakit/transaction.go]: this is indeed probably wrong, we want `len(path)` here. Or just append `path` to `p`.

Comment (talex5) [inline api/go-datakit/transaction.go]: Agree. Dropping the last element of `path` makes no sense.

Comment (talex5) [inline api/go-datakit/transaction.go]: I think you can just do `range path` on its own here to get all the elements.

**Comment by talex5** [comment]
This also explains why the bridge fails to handle PRs being deleted (the Go hooks try to delete the wrong directory). Though I'm not sure why we don't see a `Failed to Remove` message in the logs when that happens.

**Pull request merged**